storage/rocksdb/rdb_datadic.cc

```diff
@@ -3514,8 +3514,21 @@ bool Rdb_tbl_def::put_dict(Rdb_dict_manager *const dict,
                            const rocksdb::Slice &key) {
   StringBuffer<8 * Rdb_key_def::PACKED_SIZE> indexes;
   indexes.alloc(Rdb_key_def::VERSION_SIZE +
+                Rdb_key_def::TABLE_CREATE_TIMESTAMP_SIZE +
                 m_key_count * Rdb_key_def::PACKED_SIZE * 2);
-  rdb_netstr_append_uint16(&indexes, Rdb_key_def::DDL_ENTRY_INDEX_VERSION);
+
+  if (rocksdb_table_dictionary_format <
```

Comment:
It might be better if this was a switch:

```
switch (rocksdb_table_dictionary_format) {
  case ROCKSDB_DATADIC_FORMAT_INITIAL:
    rdb_netstr_append_uint16(&indexes, Rdb_key_def::DDL_ENTRY_INDEX_VERSION_1);
    set_create_time(0);
    break;
  case ROCKSDB_DATADIC_FORMAT_CREATE_TIMESTAMP:
    rdb_netstr_append_uint16(&indexes, Rdb_key_def::DDL_ENTRY_INDEX_VERSION_2);
    rdb_netstr_append_uint64(&indexes, create_time);
    break;
  default:
    DBUG_ASSERT(0);
    rdb_netstr_append_uint16(&indexes, Rdb_key_def::DDL_ENTRY_INDEX_VERSION_1);
    set_create_time(0);
    break;
}
```

This would be easier to extend. At minimum for the else {}, we should assert rocksdb_table_dictionary_format == ROCKSDB_DATADIC_FORMAT_CREATE_TIMESTAMP for this build.

Reply:
Done